x/concentrated-liquidity/math/math.go

```diff
@@ -72,8 +72,7 @@ func CalcAmount0Delta(liq, sqrtPriceA, sqrtPriceB osmomath.BigDec, roundUp bool)
 		// - calculating amountIn during swap
 		// - adding liquidity (request user to provide more tokens in in favor of the pool)
 		// The denominator is truncated to get a higher final amount.
-		denom := sqrtPriceA.MulTruncate(sqrtPriceB)
-		return liq.Mul(diff).QuoMut(denom).Ceil()
+		return liq.MulRoundUp(diff).QuoTruncate(sqrtPriceA).QuoTruncate(sqrtPriceB).Ceil()
```

Comment:
What was the reasoning behind having the divisions be QuoTruncate if we're trying to round up? My understanding is this truncates at precision and thus rounds those operations down

Reply:
Thanks for catching this! I've been staring at this way too long and inverted the direction in my head. Fixed here:
[f149a90](https://github.com/osmosis-labs/osmosis/pull/6352/commits/f149a903967ffd2094744d86f2b7eb96cf7f84ae)